### PR TITLE
refactor: move native window gin converter

### PR DESCRIPTION
**Pull request by DeerMichel**
#### Description of Change
[Continuation of an endless PR series]

I put this converter in `message_box_converter.h` at first because it was sufficient for the `dialog` migration (#19482). However, this temporary approach quickly turned out to be a bad decision for further migrations. ~So let's place it right beneath the original one.~ Now, it's moved to a new file.

cc @nornagon @deepak1556 @codebytere 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes

Notes: none

### CODE REVIEW

**Comment by deepak1556** [comment]
Also is future refactors make sure to move out converters,

* If converter is inlined in a header file
* If converter is inlined in source file, you can leave it as it is unless the same converter is to be duplicated elsewhere.

**Comment by DeerMichel** [comment]
I'm 💯 with you... this will mitigate some weird issues i'm encountering with further migrations :)

File: shell/common/native_mate_converters/file_dialog_converter.cc [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include "shell/browser/api/atom_api_browser_window.h"
```
Comment (deepak1556) [inline shell/common/native_mate_converters/file_dialog_converter.cc]: Api header shouldn't be needed now.

**Comment by DeerMichel** [comment]
huh, ill take a look

File: shell/common/native_mate_converters/file_dialog_converter.cc (hunk 1 above)
Comment (DeerMichel) [reply shell/common/native_mate_converters/file_dialog_converter.cc]: We need it in [L.65](https://github.com/electron/electron/pull/19577/files#diff-2c4b6a568b2c88fa116f70644e3fd219R65), at least my compiler says so 😄.

**Comment by DeerMichel** [comment]
All right, should be fine now 👍

**Pull request merged**